File: Java implementation/src/ChaoticIteration.java
import java.util.ArrayList;
import java.util.List;

public class ChaoticIteration extends WorkList {

    @Override
    public List<Integer> empty() {
        return new ArrayList<>();
    }

    @Override
    public void insert(List<Integer> W, int q) {
        W.add(q);
    }

    @Override
    public int extract(List<Integer> W) {
        return W.remove(0);
    }
}

File: CppUpdated/Algorithms/ChaoticIteration.cpp
#include <cstdlib>
#include <ctime>

#include "ChaoticIteration.h"

ChaoticIteration::ChaoticIteration(){
    srand((unsigned) time(0));
}

void ChaoticIteration::empty(){
    this->W = vector<int>();
}

bool ChaoticIteration::isEmpty(){
    return this->W.empty();
}

void ChaoticIteration::insert(int q) {
    this->W.push_back(q);
}

int ChaoticIteration::extract() {
    int index = rand() % this->W.size();
    int q = this->W[index];
    this->W.erase(W.begin() + index);
    return q;
}